src/Data/PreciseDateTime.purs

```diff
@@ -154,4 +174,13 @@ toDateTimeLossy :: PreciseDateTime -> DateTime
 toDateTimeLossy (PreciseDateTime dt _) = dt
 
 fromDateTime :: DateTime -> PreciseDateTime
-fromDateTime dt = PreciseDateTime dt bottom
+fromDateTime dt = PreciseDateTime dt ns
+  where
+  ms :: Millisecond
+  ms = millisecond $ time dt
+
+  nsFromMs :: Int
+  nsFromMs = fromEnum ms * 1000000
+
+  ns :: Nanosecond
+  ns = fromMaybe bottom $ toEnum nsFromMs
```

Comment:
I don't think this is necessary. Nanos should always be sub-millis. If you convert millis to nanos, you are representing it twice.

Reply:
This was a result of looking at how `adjust` uses it. I'll look into changing that.

Reply:
This was a result of looking at how `adjust` uses it. I'll look into changing that.